### PR TITLE
added data-hub-api--stg config

### PR DESCRIPTION
**Pull request by de-code**
part of https://github.com/elifesciences/data-hub-issues/issues/549

### CODE REVIEW

**Comment by de-code** [comment]
I put deployment, service and ingress into one file as that might make it easier to initially deploy staging and prod without kustomize (and we seem to have put multiple definitions in a single file in other cases).

**Comment by HazalCiplak** [comment]
I need to catch the train, not able to look too detailed, but I can see the folder name is `automation`, for consistency maybe we can change it as `automations`?

File: deployments/data-hub/data-hub-api/data-hub-api--stg.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - host: data-hub-api--stg.elifesciences.org
+      http:
+        paths:
+          - pathType: Prefix
+            path: "/"
+            backend:
+              service:
+                name: data-hub-api--stg
+                port:
+                  number: 80
```
Comment (scottaubrey) [inline deployments/data-hub/data-hub-api/data-hub-api--stg.yaml:76]: I think this needs to be port 8000?

**Review by scottaubrey — CHANGES_REQUESTED** [review]
Just a fix for the service selector, but it looks viable.

**Review by scottaubrey — APPROVED** [review]

**Pull request merged**